packages/core/admin/admin/src/content-manager/components/DynamicZone/components/DynamicComponent.js

```diff
@@ -103,80 +118,100 @@ const DynamicZoneComponent = ({
-                    })}
-                    onClick={onMoveComponentUpClick}
-                    icon={<ArrowUp />}
-                  />
-                )}
-                {isFieldAllowed && (
-                  <IconButtonCustom
-                    noBorder
-                    label={formatMessage(
-                      {
-                        id: getTrad('components.DynamicZone.delete-label'),
-                        defaultMessage: 'Delete {name}',
-                      },
-                      { name: friendlyName }
-                    )}
-                    onClick={onRemoveComponentClick}
-                    icon={<Trash />}
-                  />
-                )}
-              </Stack>
-            }
-            title={`${friendlyName}${mainValue}`}
-            togglePosition="left"
-          />
-          <AccordionContent>
-            <AccordionContentRadius background="neutral0">
-              <FieldComponent
-                componentUid={componentUid}
-                icon={icon}
-                name={`${name}.${index}`}
-                isFromDynamicZone
-              />
-            </AccordionContentRadius>
-          </AccordionContent>
-        </Accordion>
+      <StyledBox ref={composedBoxRefs} hasRadius>
+        {isDragging ? (
+          <Preview ref={dragPreviewRef} padding={6} background="primary100" />
+        ) : (
+          <Accordion expanded={isOpen} onToggle={handleToggle} size="S" error={errorMessage}>
+            <AccordionToggle
+              startIcon={<FontAwesomeIcon icon={icon} />}
+              action={
+                isFieldAllowed ? (
+                  <Stack horizontal spacing={0} expanded={isOpen}>
+                    <IconButtonCustom
+                      noBorder
+                      label={formatMessage(
+                        {
+                          id: getTrad('components.DynamicZone.delete-label'),
+                          defaultMessage: 'Delete {name}',
+                        },
+                        { name: friendlyName }
+                      )}
+                      onClick={onRemoveComponentClick}
+                    >
+                      <Trash />
+                    </IconButtonCustom>
+                    <IconButton
+                      forwardedAs="div"
+                      role="button"
+                      noBorder
+                      tabIndex={0}
+                      onClick={(e) => e.stopPropagation()}
+                      data-handler-id={handlerId}
+                      ref={dragRef}
+                      label={formatMessage({
+                        id: getTrad('components.DragHandle-label'),
+                        defaultMessage: 'Drag',
+                      })}
+                      onKeyDown={handleKeyDown}
+                    >
+                      <Drag />
+                    </IconButton>
+                  </Stack>
+                ) : null
```

Comment:
Can you elaborate please?

Reply:
I meant I'd prefer to not build the action component inline because it can be `null` which I find hard to read. Maybe there is also some small benefit in memorizing it?

```js
let ToggleAction = !isFieldAllowed ? null : <Stack>...</Stack>;
<AccordionToggle action={ToggleAction} />
```

Does that make sense?

Reply:
Ahhhh, yes, thanks it does. IMO this is why I prefer composition but that's a different debate lol